### PR TITLE
feat(vulnerabilities): Make CVE search configurable

### PR DESCRIPTION
**Pull request by bodetc**
Allows for configuring the CVE search using sw360.properties

### CODE REVIEW

**Comment by mcjaeger** [comment]
maybe, if you find it use maybe some info to the threshold value could be added. (because for the threshold it is not obvious what the threshold does) but this is not review / relevant IMHO.

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java [hunk 1]
```diff
@@ -77,48 +93,49 @@ private void setup(CveSearchApi cveSearchApi, int vendorThreshold, int productTh
...
+
     private void addCPESearchLevel() {
         Predicate<Release> isPossible = r -> r.isSetCpeid() && isCpe(r.getCpeid().toLowerCase());
         searchLevels.add(r -> {
-            if(isPossible.test(r)){
+            if (isPossible.test(r)) {
                 return singletonList(new NeedleWithMeta(cleanupCPE(r.getCpeid()), "CPE"));
             }
-            return EMPTY_LIST;
+            return ImmutableList.of();
```
Comment (alexbrdn) [inline backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: Why do you think this is better than `Collections.EMPTY_LIST`? It doesn't look like a necessary change.

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java [hunk 2]
```diff
@@ -77,48 +93,49 @@ private void setup(CveSearchApi cveSearchApi, int vendorThreshold, int productTh
...
+                            throw new UncheckedIOException(e);
+                        }
+                    });
         } catch (UncheckedIOException ue) {
             throw ue.getIOExceptionCause();
         }
     }
 
-    class UncheckedIOException extends RuntimeException{
+    class UncheckedIOException extends RuntimeException {
```
Comment (alexbrdn) [inline backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: Here and at multiple other places in this PR: unnecessary change in formatting. We have agreed not to do that. See https://github.com/sw360/sw360portal/wiki/Dev-DoD-and-Style

Please correct this first as it makes very hard to see what has been actually changed in the functionality

File: backend/src/src-cvesearch/src/test/resources/sw360.properties [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# Copyright (c) Bosch Software Innovations GmbH 2016.
+#
+# All rights reserved. This configuration file is provided to you under the
+# terms and conditions of the Eclipse Distribution License v1.0 which
+# accompanies this distribution, and is available at
+# http://www.eclipse.org/org/documents/edl-v10.php
+
+cvesearch.default.vendor.threshold=1
+cvesearch.default.product.threshold=0
+cvesearch.default.cutoff=6
```
Comment (alexbrdn) [inline backend/src/src-cvesearch/src/test/resources/sw360.properties]: missing newline at EOF

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java (hunk 1 above)
Comment (bodetc) [reply backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: It avoids an unchecked cast. It is definitely not necessary, but do we have a guideline to make only necessary changes? I am used to work with a Sonar instance running on the code, so I tend to correct the "warnings" when I see them.

File: backend/src/src-cvesearch/src/test/resources/sw360.properties (hunk 3 above)
Comment (bodetc) [reply backend/src/src-cvesearch/src/test/resources/sw360.properties]: Doesn't it count as formatting? I fixed this anyway :)

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java (hunk 2 above)
Comment (bodetc) [reply backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: I have corrected the feature in my IDE, only changed lines will be formatted. I don't see the point to revert the line that were changed by mistake back to an inconsistant formatting.

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java (hunk 1 above)
Comment (alexbrdn) [reply backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: No, no, I do not suggest to only make the absolutely necessary changes. I only wondered, what moved you to reduce the readability of the code. IMO, `EMPTY_LIST` is much easier to parse at a glance.
What about using `Collections.emptyList()`? Looks like a win-win to me.

File: backend/src/src-cvesearch/src/test/resources/sw360.properties (hunk 3 above)
Comment (alexbrdn) [reply backend/src/src-cvesearch/src/test/resources/sw360.properties]: Yes, it is formatting, obviously. But that does not count as changing someone else's formatting.

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java (hunk 2 above)
Comment (alexbrdn) [reply backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: Thing is, it's not "the line". It's a lot of lines. I'm having a hard time reviewing what you actually changed. I can't and won't make you do it, but it would be oh so kind of you if you did.

Comment (bodetc) [reply backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: I reverted most of them, it should be more readable now :)

File: backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java (hunk 1 above)
Comment (bodetc) [reply backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/heuristics/SearchLevels.java]: I changed both occurences to Collections.emptyList()

File: backend/src/src-cvesearch/src/test/resources/sw360.properties (hunk 3 above)
Comment (bodetc) [reply backend/src/src-cvesearch/src/test/resources/sw360.properties]: done.
